Userland/Libraries/LibTextCodec/Decoder.cpp

```diff
@@ -372,7 +372,7 @@ ErrorOr<String> UTF8Decoder::to_utf8(StringView input)
         bomless_input = input.substring_view(3);
     }
 
-    return Decoder::to_utf8(bomless_input);
+    return String::from_utf8(bomless_input);
```

Comment:
I was wondering about this when I changed the UTF-16 decoders. I think this prevents invalid UTF-8 from getting the replacement char treatment, which is required by the spec:
https://encoding.spec.whatwg.org/#utf-8-decode

Reply:
Maybe we can do something like:
```c++
if (Utf8View { bomless_input }.validate())
    return String::from_utf8_without_validation(bomless_input);
return Decoder::to_utf8(bomless_input);
```

Reply:
Yes, let's do that at least!